pkg/apis/core/validation/validation.go

```diff
@@ -5311,6 +5316,15 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
 		}
 	}
 
+	if windowsOptions.RunAsUserName != nil {
+		if l := len(*windowsOptions.RunAsUserName); l == 0 {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("runAsUserName"), windowsOptions.RunAsUserName, "runAsUserName cannot be an empty string"))
+		} else if l >= maxRunAsUserNameLength {
+			errMsg := fmt.Sprintf("runAsUserName's length must be under %d characters", maxRunAsUserNameLength)
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("runAsUserName"), windowsOptions.RunAsUserName, errMsg))
+		}
+	}
```

Comment:
are there any dangerous characters or sequences we need to be concerned about (backslashes, escapes, `..`, etc)?

Reply:
I finally found a relevant up-to-date [reference](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.localaccounts/new-localuser?view=powershell-5.1) that specifies the following:

```
If you create a local user account for the local system, the user name can contain up to 20 uppercase characters or lowercase characters. A user name cannot contain the following characters:

" / \ [ ] : ; | = , + * ? < > @

A user name cannot consist only of periods (.) or spaces.
```

It seems in Server 2019, `New-LocalUser` did not complain when passed `u..r`. So I think `..` is allowed in accordance with the above. `New-LocalUser`  did complain when passed both `u/r` and `u\r` as usernames.

It appears the above list also appears in docs from Windows 2000 Active Directory [restrictions](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-2000-server/bb726984(v=technet.10)) with the exception of `@`. 

So it may be safe to reject characters: `" / \ [ ] : ; | = , + * ? < >` without being too restrictive from a future compatibility perspective? /cc @PatrickLang

Reply:
Good point. We do need to allow domain prefixed names like `NT AUTHORITY\NETWORK SERVICE` through.

Reply:
Done